### PR TITLE
AWS required permission

### PR DESCRIPTION
**Pull request by yunjiang29**
* New workflow `cucushift-installer-rehearse-aws-ipi-minimal-permission`
* Rename `aws-deprovision-cco-manual-users-static ` to `aws-deprovision-users-and-policies `

### CODE REVIEW

File: ci-operator/step-registry/aws/provision/iam-user-with-required-permission/aws-provision-iam-user-with-required-permission-commands.sh [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+s3:PutLifecycleConfiguration
+s3:PutObject
+s3:PutObjectAcl
+s3:PutObjectTagging
+servicequotas:ListAWSDefaultServiceQuotas
+tag:GetResources
+EOF
+
+# additional permisions for 4.11+
+if (( ocp_minor_version >= 11 && ocp_major_version >= 4 )); then
```
Comment (jianlinliu) [inline ci-operator/step-registry/aws/provision/iam-user-with-required-permission/aws-provision-iam-user-with-required-permission-commands.sh:248]: How about `ocp_major_version == 4` to be more restrict.

File: ci-operator/step-registry/aws/provision/iam-user-with-required-permission/aws-provision-iam-user-with-required-permission-ref.metadata.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"path": "aws/provision/iam-user-with-required-permission/aws-provision-iam-user-with-required-permission-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/aws/provision/iam-user-with-required-permission/aws-provision-iam-user-with-required-permission-ref.metadata.json:2]: suggest rename the path like `aws/provision/iam-user/minimal-permission`

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/required-permission/cucushift-installer-rehearse-aws-ipi-required-permission-workflow.metadata.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+	"path": "cucushift/installer/rehearse/aws/ipi/byo-kms/etcd-encryption/cucushift-installer-rehearse-aws-ipi-byo-kms-etcd-encryption-workflow.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/required-permission/cucushift-installer-rehearse-aws-ipi-required-permission-workflow.metadata.json:2]: copy/paste error?

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/required-permission/deprovision/cucushift-installer-rehearse-aws-ipi-required-permission-deprovision-chain.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+chain:
+  as: cucushift-installer-rehearse-aws-ipi-required-permission-deprovision
+  steps:
+  - chain: cucushift-installer-rehearse-aws-ipi-deprovision
+  - ref: aws-deprovision-cco-manual-users-static
```
Comment (yunjiang29) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/required-permission/deprovision/cucushift-installer-rehearse-aws-ipi-required-permission-deprovision-chain.yaml:5]: @jianlinliu how about renaming this step to a more generic name, like `aws-deprovision-users-and-policies`

Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/required-permission/deprovision/cucushift-installer-rehearse-aws-ipi-required-permission-deprovision-chain.yaml:5]: yeah, agree.

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

File: ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh [hunk 5]
```diff
@@ -13,7 +13,13 @@ trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wa
 trap 'save_logs' EXIT TERM
 
 export ALIBABA_CLOUD_CREDENTIALS_FILE=${SHARED_DIR}/alibabacreds.ini
-export AWS_SHARED_CREDENTIALS_FILE=$CLUSTER_PROFILE_DIR/.awscred
+if [[ -f "${SHARED_DIR}/aws_minimal_permission" ]]; then
+  echo "Setting AWS credential with minimal permision for installer"
+  export AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/aws_minimal_permission
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-commands.sh:18]: `SHARED_DIR` here, right?

File: ci-operator/step-registry/aws/deprovision/users-and-policies/aws-deprovision-users-and-policies-ref.yaml [hunk 6]
```diff
@@ -1,11 +1,11 @@
 ref:
-  as: aws-deprovision-cco-manual-users-static
```
Comment (jianlinliu) [inline ci-operator/step-registry/aws/deprovision/users-and-policies/aws-deprovision-users-and-policies-ref.yaml:2]: Seem like ci-operator/step-registry/quay-tests/aws-ipi-disconnected-private-deprovision/quay-tests-aws-ipi-disconnected-private-deprovision-chain.yaml also reference the step. cc @LiZhang19817

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-installer-rehearse-aws

**Comment by yunjiang29** [comment]
@LiZhang19817 can you review the changes on `quay-tests-aws-ipi-disconnected-private-deprovision-chain.yaml`?  thanks

**Comment by yunjiang29** [comment]
@liangxia PTAL, thanks

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml [hunk 7]
```diff
@@ -326,6 +326,17 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-aws-ipi-disconnected-private-workers-rhcos-rhel8
+- as: aws-ipi-minimal-permission-p3-f28
+  cron: 22 18 13 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      E2E_RUN_TAGS: '@amd64 and @aws-ipi'
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml:335]: We might need the network type and not fips, too

Comment (yunjiang29) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.10__amd64-nightly.yaml]: fixed, thanks

**Pull request merged**